dig_test.go

```diff
@@ -462,6 +462,35 @@ func TestEndToEndSuccess(t *testing.T) {
 			require.NotNil(t, a, "*A should be part of the container through Ret2->Ret1")
 		}))
 	})
+
+	t.Run("named instances", func(t *testing.T) {
```

Comment:
More tests are in order? I'll think about circular dependency resolution(named instances could have a different dependency graphs that are valid, but could look like a loop if you drop names), what should we return if there is only a named instance(or two) for the unnamed type and reverse.